cloudferrylib/os/actions/cinder_database_manipulation.py

```diff
@@ -173,6 +175,76 @@ def __init__(self, *args, **kwargs):
         self.dst_volume_types = None
         self.path_map = None
         self.mount_all = {}
+        self.img_map = None
+
+    def _skip_existing_volumes(self):
+        res = []
+        dst_ids = [v['id'] for v in self.data[DST]['volumes']]
+        for v in self.data[SRC]['volumes']:
+            if v['id'] in dst_ids:
+                LOG.warning('Volume %s existing, skipping migration.', v['id'])
+            else:
+                res.append(v)
+        self.data[SRC]['volumes'] = res
+
+    def read_images_info(self, pos):
+        """Read images info.
+
+        :return: list
+
+        """
+        res = self.cloud[pos][CLOUD].resources.get(utils.IMAGE_RESOURCE)
+        return res.read_info()['images'].values()
+
+    def _image_map(self, img_id):
+        if self.img_map is not None:
+            return self.img_map.get(img_id, None)
+
+        self.img_map = {}
+
+        imgs, keystone = {}, {}
+        for pos in SRC, DST:
+            keystone[pos] = self.cloud[pos][CLOUD].resources.get(
+                utils.IDENTITY_RESOURCE)
+            imgs[pos] = self.read_images_info(pos)
+
+        for img in imgs[SRC]:
+            for dst_img in imgs[DST]:
+                skip_fields = ['id', 'resource', 'owner']
+                for f in img['image'].keys():
+                    if f not in skip_fields and \
+                            img['image'][f] != dst_img['image'][f]:
+                        break
+                    if f == 'owner' and \
+                            keystone[SRC].try_get_tenant_name_by_id(
+                                img['image'][f]) != \
+                            keystone[DST].try_get_tenant_name_by_id(
+                                dst_img['image'][f]):
+                        break
+                else:
+                    self.img_map[img['image']['id']] = dst_img['image']['id']
```

Comment:
Please move this logic to glance and rename to something like `glance_images_identical(src_image, dst_image)`